adversarial_text/data/data_utils.py

```diff
@@ -215,13 +215,16 @@ def build_lm_sequence(seq):
 
   Returns:
     SequenceWrapper with `seq` tokens copied over to output sequence tokens and
-    labels (offset by 1, i.e. predict next token) with weights set to 1.0.
+    labels (offset by 1, i.e. predict next token) with weights set to 1.0, except for <eos> token.
   """
   lm_seq = SequenceWrapper()
-  for i, timestep in enumerate(seq[:-1]):
-    lm_seq.add_timestep().set_token(timestep.token).set_label(
+  for i, timestep in enumerate(seq):
```

Comment:
So given a seq ['a', 'b', 'c', 'eos'], the old behavior would have produced:
[(a, b, 1.0), (b, c, 1.0), (c, eos, 1.0)]
where each entry is (token, label, weight).
The new behavior produces:
[(a, b, 1.0), (b, c, 1.0), (c, eos, 1.0), (eos, eos, 0.0)].

Why is that last EOS necessary?